### PR TITLE
Task/update attributes in unexistant array

### LINKED ISSUE #277: Updating device provisioning for new attributes may not be working.
Using this curl script for device providing:

```
#!/bin/bash

(curl localhost:4041/iot/devices -s -S --header 'Content-Type: application/json' \
  --header 'Accept: application/json' --header 'fiware-service: ' --header 'fiware-servicepath: /mqtt' \
  -d @- | python -mjson.tool) <<EOF
{
  "devices": [
    {
      "device_id": "MQTT",
      "protocol": "GENERIC_PROTO",
      "entity_name": "MQTT Device",
      "entity_type": "Device",
      "timezone": "America/Santiago"
    }
  ]
}
EOF
```

And then this for device updating:

```
#!/bin/bash

(curl localhost:4041/iot/devices/MQTT -s -S -X PUT --header 'Content-Type: application/json' \
  --header 'Accept: application/json' --header 'fiware-service: ' --header 'fiware-servicepath: /mqtt' \
  -d @- | python -mjson.tool) <<EOF
{
  "attributes": [
    {
        "name": "nuevoAtributo",
        "type": "Integer"
    }
  ]
}
EOF
```

With MQTT IoTAgent (which uses this library) triggers this:

```
time=2016-02-16T09:36:00.090Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Body:

{
    "attributes": [
        {
            "name": "nuevoAtributo",
            "type": "Integer"
        }
    ]
}


time=2016-02-16T09:36:00.093Z | lvl=DEBUG | corr=n/a | trans=n/a | op=IoTAgentNGSI.NorthboundServer | msg=Error [TypeError] handing request: Cannot read property '0' of undefined
```

Any idea?

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #277

### CODE REVIEW

**Comment by XavierVal** [comment]
LGTM

**Comment by anabelengp** [comment]
LGTM

File: test/unit/contextRequests/updateActiveAttributes.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "contextElements": [
+    {
+      "type": "TheLightType",
+      "isPattern": "false",
+      "id": "TheFirstLight",
+      "attributes": []
+    }
+  ],
+  "updateAction": "APPEND"
+}
```
Comment (gtorodelvalle) [inline test/unit/contextRequests/updateActiveAttributes.json]: Final blank line appreciated... :)

File: test/unit/contextRequests/updateProvisionActiveAttributes1.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        {
+          "name": "temperature",
+          "type": "centigrades",
+          "value": " "
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}
```
Comment (gtorodelvalle) [inline test/unit/contextRequests/updateProvisionActiveAttributes1.json]: Final blank line appreciated... ;)

File: test/unit/contextResponses/updateActiveAttributesSuccess.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        "isPattern": "false",
+        "type": "TheLightType"
+      },
+      "statusCode": {
+        "code": "200",
+        "reasonPhrase": "OK"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/updateActiveAttributesSuccess.json]: Final blank line appreciated... ;)

File: test/unit/contextResponses/updateProvisionActiveAttributes1Success.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        "isPattern": "false",
+        "type": "Light"
+      },
+      "statusCode": {
+        "code": "200",
+        "reasonPhrase": "OK"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/updateProvisionActiveAttributes1Success.json]: Final blank line appreciated... ;)

File: test/unit/contextResponses/updateProvisionMinimumDeviceSuccess.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        "isPattern": "false",
+        "type": "MicroLights"
+      },
+      "statusCode": {
+        "code": "200",
+        "reasonPhrase": "OK"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/updateProvisionMinimumDeviceSuccess.json]: FInal blank line appreciated... ;)

File: test/unit/deviceProvisioningRequests/updateMinimumDevice.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "attributes": [
+    {
+      "name": "newAttribute",
+      "type": "Integer"
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/deviceProvisioningRequests/updateMinimumDevice.json]: Final blank line appreciated... ;)

**Comment by gtorodelvalle** [comment]
LGTM and merging... ;)

**Pull request merged**